### DISCUSSION TITLE
Q: vue项目引入@waline/client报错

### DISCUSSION THREAD

**Discussion by elainecyx** [walinejs/waline#1941] [Q&A] [answered]
按照官方文档提供的方式引入：

```
 npm i -D @waline/client

<template>
<Waline :serverURL="serverURL" :path="path" />
</template>
<script setup>
import { Waline } from '@waline/client/component';
import { computed } from 'vue';
import { useRoute } from 'vue-router';

const serverURL = 'https://waline.vercel.app';
const path = computed(() => useRoute().path);
</script>
```

有报错，报错信息如下：
This dependency was not found:
* @waline/client/component in ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/views/userManagement/user/userList.vue?vue&type=script&lang=js&

To install it, you can run: npm install --save @waline/client/component

依赖的版本号：
"@waline/client": "^2.15.0"
"vue": "2.6.10"

**Comment by lizheming** [comment, marked as answer]
`@waline/client` 基于 Vue3 开发的，不支持组件形式 Vue2 引入。

**Reply by elainecyx** [reply]
基于nuxtjs框架下,https://nuxtjs.org/docs/get-started/installation
```
"@nuxtjs/axios": "^5.13.6",
"@nuxtjs/dayjs": "^1.4.1",
"@nuxtjs/google-fonts": "^2.0.0",
"@nuxtjs/i18n": "^7.2.0",
"vue": "^2.6.14"
```
使用以下方式引入：
```
import { init } from '@waline/client';

import '@waline/client/dist/waline.css';

init({
  el: '#waline',
  // ...
});
```
`node版本： v14.21.3`

出现报错：
```
ERROR in ./node_modules/@waline/client/dist/legacy.umd.js 1:57040
Module parse failed: Unexpected token (1:57040)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
```
![image](https://user-images.githubusercontent.com/79910670/232956094-089fba45-3acf-477c-9a01-170f516aed52.png)

**Reply by lizheming** [reply]
升级下 Node 版本

**Reply by elainecyx** [reply]
![image](https://user-images.githubusercontent.com/79910670/232963083-23678b6e-1c36-47a2-bbfb-5cdeec65b35f.png)

node版本： v18.16.0

**Reply by Mister-Hope** [reply]
The following workaround may help, but it's just a workaround (I am not sure whether webpack4 supports "exports" field in package.json)

```diff
- import { init } from '@waline/client';
+ import { init } from '@waline/client/full';
```

You are expected to resolve any outdated deps and env issues yourself if you want to use them. This is out of scope, as we are only willing to help newcomers or noobs who have just started learning. There is no tech support here like the way you are asking now.

Vue2 webpack4 are all outdated deps. If you use the latest nuxt3 and provide reproduction and troubles you meet, we would like to help.

**Reply by lizheming** [reply]
不太清楚了，Waline 应该是不支持 SSR 的，你就不应该在 Nuxt SSR 部分去加载 Waline，应该放到 Next 客户端部分进行渲染。要是不会使用建议别折腾了吧。@Mister-Hope hopes 老师帮忙确认下？看看有没有啥指导性建议

**Reply by Mister-Hope** [reply]
> 不太清楚了，Waline 应该是不支持 SSR 的，你就不应该在 Nuxt SSR 部分去加载 Waline，应该放到 Next 客户端部分进行渲染。要是不会使用建议别折腾了吧。@Mister-Hope hopes 老师帮忙确认下？看看有没有啥指导性建议
> 

原则上可以。没有什么不是不行的。我一直在确认waline的ssr支持。他唯一要做的是处理好path。

评论和访问数量的确不行，因为他们涉及到外部dom获取与修改。

**Reply by elainecyx** [reply]
SSR应该是不影响，需要把waline引入放到client端？为了确定不是SSR的影响，我先在vue2项目做了测试，目前按照@Mister-Hope 使用以下导入是OK的，没有报依赖的问题。
import { init } from '@waline/client/full';
可以成功看到comment box, 如图，
![image](https://user-images.githubusercontent.com/79910670/232985813-0b116cd7-903c-46d6-a852-1143c4d3d1e0.png)

不过点击refresh，出现错误：
` WalineComment.vue?7ffc:193 Failed to execute 'fetch' on 'Window': Failed to parse URL from https://http:127.0.0.1:8360/comment?path=%2F&pageSize=10&page=1&lang=en&sortBy=insertedAt_desc`

**Reply by elainecyx** [reply]
不好意思，我这边正常了，serverURL格式写错了，感谢两位大神的帮忙！我再引入到nuxtjs看下情况。

**Reply by Mister-Hope** [reply]
## DOUBLE CHECK HOW YOU GET: `http:127.0.0.1:8360` if this is set by yourself, this should be `127.0.0.1:8360` or `http://127.0.0.1:8360`

**Reply by elainecyx** [reply]
在nuxtjs以同样的方式引入依赖 `import { init } from '@waline/client/full';`会出现报错，目前是在client端在mounted中初始化waline
报错信息如下：
`Failed to compile with 1 errors                                                                           friendly-errors 14:56:00

This dependency was not found:                                                                                    friendly-errors 14:56:00
                                                                                                                  friendly-errors 14:56:00
* @waline/client/full in ./node_modules/babel-loader/lib??ref--2-0!./node_modules/vue-loader/lib??vue-loader-options!./node_modules/string-replace-loader??ref--12!./pages/acg/detail/_id.vue?vue&type=script&lang=js&
                                                                                                                  friendly-errors 14:56:00
To install it, you can run: npm install --save @waline/client/full `

**Reply by elainecyx** [reply]
nuxtjs已解决！
分享给大家解决方案：
nuxt.config.js在head声明以下内容：
```
head: {
    link: [
      { rel: 'stylesheet', href: 'https://unpkg.com/@waline/client@v2/dist/waline.css' }
    ],
    script: [
      {
        src: 'https://unpkg.com/@waline/client@v2/dist/waline.js',
        type: 'text/javascript',
        pbody: true,
        defer: true
      }
    ]
  }
```
在xxx.vue页面中
`<template><div id="waline" /></template>`

在mounted生命周期
```
const params = new URLSearchParams(location.search.slice(1))
  /* eslint-disable-next-line no-undef */
  return Waline.init({
    el: '#waline',
    path: params.get('path') || '/',
    lang: params.get('lng'),
    serverURL: 'your server url'
  })

```

**Reply by Apcin** [reply]
@elainecyx 你好可以问你一下有关项目的问题吗


---
@elainecyx Hello, can I ask you a question about the project?
